### PR TITLE
bumps (library-go)

### PR DESCRIPTION
**Pull request by p0lyn0mial**
picks up https://github.com/openshift/library-go/pull/926 which sets `TCP_USER_TIMEOUT` socket option for `KAS - OAS`

### CODE REVIEW

File: pkg/image/apis/image/dockerpre012/conversion.go [hunk 1]
```diff
@@ -11,10 +11,10 @@ import (
 
 // Convert docker client object to internal object, but only when this package is included
 func Convert_dockerpre012_ImagePre_012_to_api_DockerImage(in *dockerpre012.ImagePre012, out *newer.DockerImage, s conversion.Scope) error {
-	if err := s.Convert(in.Config, &out.Config); err != nil {
+	if err := s.Convert(in.Config, &out.Config, conversion.DestFromSource); err != nil {
```
Comment (p0lyn0mial) [inline pkg/image/apis/image/dockerpre012/conversion.go:14]: I think that `conversion.DestFromSource` was the default.

File: pkg/image/apis/image/v1/zz_generated.conversion.go [hunk 2]
```diff
@@ -381,7 +381,7 @@ func autoConvert_v1_Image_To_image_Image(in *v1.Image, out *image.Image, s conve
 	out.ObjectMeta = in.ObjectMeta
 	out.DockerImageReference = in.DockerImageReference
 	// TODO: Inefficient conversion - can we improve it?
-	if err := s.Convert(&in.DockerImageMetadata, &out.DockerImageMetadata); err != nil {
+	if err := s.Convert(&in.DockerImageMetadata, &out.DockerImageMetadata, 0); err != nil {
```
Comment (p0lyn0mial) [inline pkg/image/apis/image/v1/zz_generated.conversion.go:384]: automation set it to `conversion.DestFromSource` as well.

File: pkg/image/apis/image/dockerpre012/conversion.go (hunk 1 above)
Comment (p0lyn0mial) [reply pkg/image/apis/image/dockerpre012/conversion.go:14]: actually - https://github.com/openshift/openshift-apiserver/pull/150/files

**Comment by p0lyn0mial** [comment]
requires https://github.com/openshift/api/pull/776

File: pkg/image/apis/image/v1/conversion.go [hunk 3]
```diff
@@ -162,7 +162,7 @@ func Convert_image_ImageStreamSpec_To_v1_ImageStreamSpec(in *newer.ImageStreamSp
 		}
 	}
 	out.Tags = make([]v1.TagReference, 0, 0)
-	return s.Convert(&in.Tags, &out.Tags)
+	return s.Convert(&in.Tags, &out.Tags, conversion.DestFromSource)
```
Comment (sttts) [inline pkg/image/apis/image/v1/conversion.go:165]: what do these mean?

File: hack/openapi-violation.list [hunk 4]
```diff
@@ -210,6 +210,7 @@ API rule violation: list_type_missing,github.com/openshift/api/oauth/v1,OAuthCli
 API rule violation: list_type_missing,github.com/openshift/api/oauth/v1,OAuthClient,ScopeRestrictions
 API rule violation: list_type_missing,github.com/openshift/api/oauth/v1,OAuthClientAuthorization,Scopes
 API rule violation: list_type_missing,github.com/openshift/api/oauth/v1,ScopeRestriction,ExactValues
+API rule violation: list_type_missing,github.com/openshift/api/oauth/v1,UserOAuthAccessToken,Scopes
```
Comment (sttts) [inline hack/openapi-violation.list:213]: @marun see here. This is a good start. We should not have these violations.

File: pkg/image/apis/image/v1/conversion.go (hunk 3 above)
Comment (p0lyn0mial) [reply pkg/image/apis/image/v1/conversion.go:165]: these were removed in https://github.com/openshift/openshift-apiserver/pull/150/files#diff-476a993839cca71a75d722226b54f2a2e7560296678290d40054fa9b009bfef3L14 

my pr just revert these changes as they were not meant to be merged.

Comment (p0lyn0mial) [reply pkg/image/apis/image/v1/conversion.go:165]: we should revert https://github.com/openshift/openshift-apiserver/pull/150/files#diff-bc324103be9275286883e8cb4282105288549a542d171c6d9abb4910475f5f92L531 as well

Comment (p0lyn0mial) [reply pkg/image/apis/image/v1/conversion.go:165]: done.

**Pull request merged**